DOCS-840: Document the Base Remote Control service (#1395)
viamrobotics/docs: docs/services/base-rc/_index.md

4. `"joyStickControl"`: One joystick controls speed and angle
5. `"droneControl"`: Two joysticks control speed and angle

You can monitor the input from these controls in the **Control** tab of the [Viam app](https://app.viam.com).

## Configuration

You must configure a [base](/components/base/) with a [movement sensor](/components/movement-sensor/) as part of your robot to be able to use a Base Remote Control service.

{{< tabs >}}
{{% tab name="Config Builder" %}}

Navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com).
Click on the **Services** subtab and navigate to the **Create service** menu.
Select the type `Navigation` and enter a name for your service.

Click **Create service**:

![An example configuration for a Base Remote Control service in the Viam app Config Builder.](img/base-rc-ui-config.png)

{{% /tab %}}
{{% tab name="JSON Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
    "name": "<your-base-remote-control-service>",
    "type": "base_remote_control",
    "attributes": {
        "base": "<your-base-name>",
        "input_controller": "<your-controller-name>"
    }
}
```

{{% /tab %}}

> {{< /tabs >}}

Reviewer: I think good to add an example as well

---
title: "Base Remote Control Service"
linkTitle: "Base Remote Control"

> weight: 80

Reviewer: Let's put this in alphabetical order (aside from robot - I think that one can stay at the top since it makes sense as the first one)

Author: Done!

type: "docs"

> description: "The Base Remote Control Service implements remote control for a base."

Reviewer: ```suggestion
description: "The Base Remote Control Service allows you to remotely control a base with an input controller like a gamepad."
```

tags: ["base", "services", "rover", "input controller", "remote control"]

> # icon: "/services/img/icons/base-remote-control.svg"

Author: @npentrel I have reached out to her!

# images: ["/services/img/icons/base-remote-control.svg"]
# SME: Eric
---

The Base Remote Control service implements an [input controller](/components/input-controller/) as a remote control for a [base](/components/base/).
This uses the [`input` api](/components/input-controller/#api) to make it easy to add remote drive controls for your rover or other mobile robot with a controller like a gamepad.

Add the Base Remote Control service after configuring your robot with a base and input controller to control the linear and angular velocity of the base with the controller's button or joystick controls.

Control mode is determined by the configuration attribute `"mode"`, for which there are five options:

1. `"arrowControl"`: Arrow buttons control speed and angle
2. `"triggerSpeedControl"`: Trigger button controls speed and joystick controls angle
3. `"buttonControl"`: Four buttons (usually X, Y, A, B) control speed and angle
4. `"joyStickControl"`: One joystick controls speed and angle
5. `"droneControl"`: Two joysticks control speed and angle

Monitor the input from these controls in the **Control** tab of the [Viam app](https://app.viam.com).

## Configuration

> You must configure a [base](/components/base/) with a [movement sensor](/components/movement-sensor/) as part of your robot to configure a Base Remote Control service.

Reviewer: ```suggestion
You must configure a [base](/components/base/) with a [movement sensor](/components/movement-sensor/) as part of your robot to be able to use a Base Remote Control service.
```

{{< tabs >}}
{{% tab name="Config Builder" %}}

Navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com).
Click on the **Services** subtab and navigate to the **Create service** menu.
Select the type `Navigation` and enter a name for your service.

Click **Create service**:

![An example configuration for a Base Remote Control service in the Viam app Config Builder.](img/base-rc-ui-config.png)

{{% /tab %}}

> {{% tab name="JSON Template" %}}

Reviewer: Please follow the style we use for other templates

Author: My bad, I think done!

Reviewer: yep perfect!

```json {class="line-numbers linkable-line-numbers"}
{
    "name": "your-base-remote-control-service",
    "type": "base_remote_control",
    "attributes": {
        "base": "your-base",
        "input_controller": "your-controller",
        "mode": "arrowControl"
    }
}
```

{{% /tab %}}
{{< /tabs >}}

Next, add the JSON `"attributes"` you want the service to have.
The following attributes are available for Base Remote Control services:

| Name | Type | Inclusion | Description |
| ---- | ---- | --------- | ----------- |
| `base` | string | **Required** | The `name` of the [base](/components/base/) you have configured for the base you are operating with this service. |
| `input_controller` | string | **Required** | The `name` of the [input controller](/components/input-controller/) you have configured for the base you are operating with this service. |

> | `control_mode` | string | Optional | The mode of remote control you want to use. <br> Options: `["arrowControl", "triggerSpeedControl", "buttonControl", "joyStickControl", "droneControl"]` <br> Default: `"arrowControl"` |

Reviewer: I think this should look slightly better
```suggestion
| `control_mode` | string | Optional | The mode of remote control you want to use. <br> Options: <ul><li>`"arrowControl"`</li><li>`"triggerSpeedControl"`</li><li>`"buttonControl"`</li><li>`"joyStickControl"`</li> <li>`"droneControl"`</li></ul> <br> Default: `"arrowControl"` |
```

| `max_angular_degs_per_sec` | float | Optional | The max angular velocity for the [base](/components/base/) in degrees per second. |
| `max_linear_mm_per_sec` | float | Optional | The max linear velocity for the [base](/components/base/) in meters per second. |

## API

> The Base Remote Control service supports the following methods:

Reviewer: ```suggestion
The Base Remote Control Service supports the following methods:
```

Method Name | Description
----------- | -----------
[`Close`](#close) | Close out of all remote control related systems.

> [`ControllerInputs`](#controllerinputs) | Get a list of inputs from the controller that are being monitored for that control mode.

Reviewer: controllers? or is?
```suggestion
[`ControllerInputs`](#controllerinputs) | Get a list of inputs from the controller that is being monitored for that control mode.
```

Author: Sounds good, done

Author: done!

{{< tabs >}}
{{% tab name="Config Builder" %}}

Navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com).
Click on the **Services** subtab and navigate to the **Create service** menu.
Select the type `Navigation` and enter a name for your service.

Click **Create service**:

![An example configuration for a Base Remote Control service in the Viam app Config Builder.](img/base-rc-ui-config.png)

{{% /tab %}}
{{% tab name="JSON Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
    "name": "<your-base-remote-control-service>",
    "type": "base_remote_control",
    "attributes": {
        "base": "<your-base-name>",
        "input_controller": "<your-controller-name>"
    }
}
```

{{% /tab %}}
{{% tab name="JSON Example" %}}
```json {class="line-numbers linkable-line-numbers"}
{
    "name": "my-base-remote-control-service",
    "type": "base_remote_control",
    "attributes": {
        "base": "my-base",
        "input_controller": "my-input-controller",
        "control_mode": "arrowControl"
    }

> }

Author: ```suggestion
}
```
```
